specifyweb/stored_queries/format.py

```diff
@@ -89,96 +101,125 @@ def pseudo_sprintf(self, format, expr):
+                                         getattr(table, specify_field.name))
+
+
+            if 'format' in fieldNodeAttrib:
+                expr = self.pseudo_sprintf(fieldNodeAttrib['format'], expr)
 
-            if 'format' in fieldNode.attrib:
-                expr = self.pseudo_sprintf(fieldNode.attrib['format'], expr)
 
-            if 'sep' in fieldNode.attrib:
-                expr = concat(fieldNode.attrib['sep'], expr)
-            return query, blank_nulls(expr)
+            if 'sep' in fieldNodeAttrib:
+                expr = concat(fieldNodeAttrib['sep'], expr)
+            return query, blank_nulls(expr), formatter_field_spec
 
-        def make_case(query: QueryConstruct, caseNode: Element) -> Tuple[QueryConstruct, Optional[str], blank_nulls]:
+        def make_case(query: QueryConstruct, caseNode: Element) -> Tuple[
+            QueryConstruct, Optional[str], blank_nulls]:
             field_exprs = []
             for node in caseNode.findall('field'):
-                query, expr = make_expr(query, node)
+                query, expr, _ = make_expr(query, node.text, node.attrib)
                 field_exprs.append(expr)
 
             expr = concat(*field_exprs) if len(field_exprs) > 1 else field_exprs[0]
-            return query, case_value_convert(caseNode.attrib.get('value', None)), expr
+            return query, caseNode.attrib.get('value', None), expr
 
+        switchNode = formatterNode.find('switch')
+        single = switchNode.attrib.get('single', 'true') == 'true'
         cases = []
         for caseNode in switchNode.findall('fields'):
             query, value, expr = make_case(query, caseNode)
             cases.append((value, expr))
 
         if not cases:
-            logger.warn("dataobjformatter for %s contains switch clause no fields", specify_model)
+            logger.warn(
+                "dataobjformatter for %s contains switch clause no fields",
+                specify_model)
             return query, literal(_("<Formatter not defined.>"))
 
         if single:
             value, expr = cases[0]
         else:
-            control_field = getattr(orm_table, switchNode.attrib['field'])
-            expr = case(cases, control_field)
-
+            query, formatted, swith_field_spec = make_expr(query, switchNode.attrib['field'], {})
+            def case_value_convert(value): return value == 'true' if swith_field_spec.get_field().type == 'java.lang.Boolean' else value
+            cases = [(case_value_convert(value), expr) for (value, expr) in cases]
+            expr = case(cases, formatted)
         return query, blank_nulls(expr)
 
-    def aggregate(self, query: QueryConstruct, field: Union[Field, Relationship] , rel_table: SQLTable, aggregator_name) -> ScalarSelect:
-        
-        logger.info('aggregating field %s on %s using %s', field, rel_table, aggregator_name)
+    def aggregate(self, query: QueryConstruct,
+                  field: Union[Field, Relationship], rel_table: SQLTable,
```

Comment:
weird formatting here. can you make each attribute start on a new line, just like prettier would do that in javascript?

Reply:
I'm not entirely sure, but I believe @realVinayak formatted the document with PEP 8 (the Python standard for formatting and style). 
If that is the case, then this change was done 'automatically'. 

@realVinayak which formatter did you use to format the document with? 
Or was it all done by hand? :grin:

Reply:
Lol it was done by PEP 8. I can revert the formatting if that's a problem.

Reply:
Not a problem, just very strange - i.e, why does it have 2 arguments on first line and then 2 arguments on the next line rather than one argument per line (like in js) - that seems far more readable

Probably I'm just not used to the intricacies of the python world